### DISCUSSION TITLE
:has(.myclass) does not seem to work

### DISCUSSION THREAD

**Discussion by doompadee** [tailwindlabs/tailwindcss#9848] [Help] [answered]
I'm trying to customize one specific framework dialog and wanted to employ the `:has` pseudo selector for filtering whether a specific class is present, but unfortunately I could not get it to work. 

Minimal example: https://play.tailwindcss.com/QIPfJEbbdi?file=css

Both boxes should be green! Interestingly `:not` works as expected. Safelisting does not make a difference. Am I doing something wrong or is this is bug?

**Comment by wongjn** [comment, marked as answer]
You are doing something wrong. The selectors within the `:has()` pseudo-class that <q>don't begin with an explicit [combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors#combinators) have an implied [descendant combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/Descendant_combinator).</q>[^1] So your rule will match `.box` when it has a child element with `custom` class on it, not when it has `custom` class on itself. If you would like the rule to apply when `.box` has the `custom` class on itself, you could combine classes like `.box.custom` or use the `:is()` pseudo-class like `.box:is(.custom)`.

[^1]: https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors#relative_selector

**Reply by doompadee** [reply]
Silly me! Using `:is` got things working as expected (of course).

Thanks very much for your quick support. Have a great day!
